### PR TITLE
Add onnx test for batched_nms

### PR DESCRIPTION
**Pull request by NicolasHug**
As requested by @jiafatom https://github.com/pytorch/vision/pull/3483#pullrequestreview-602124680

### CODE REVIEW

**Review by jiafatom — APPROVED** [review]
This PR is in sync with the merged PR https://github.com/pytorch/vision/pull/3483/

(1) Please merge the PR into pytorch:onnx_ms_1, not pytorch:master. 
onnx_ms_1 is our branch to collect PRs (then we will merge to master).
(2) the ort_test2 test failure seems related, please fix it. Thanks.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -6365,16 +6365,32 @@ def forward(self, x, y):
...
+        boxes = torch.rand(num_boxes, 4)
+        boxes[:, 2:] += boxes[:, :2]
+        scores = torch.randn(num_boxes)
+        idxs = torch.randint(0, 5, size=(num_boxes,))
+
+        class Module(torch.nn.Module):
+            def forward(self, boxes, scores, idxs):
+                return ops.batched_nms(boxes, scores, idxs, 0.5)
+
+        self.run_model(Module(), [(boxes, scores, idxs)])
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:6392]:
```suggestion
        self.run_test(Module(), (boxes, scores, idxs))
```

**Review by BowenBao — APPROVED** [review]
Thanks!

**Comment by garymm** [comment]
I'm trying to run the test locally on top of the latest onnx_ms_1 and running into surprising failures. For example `TestONNXRuntime_opset14.test_nms` fails with:
```                                                                                                                                                                                                                                                                                                          
packages/onnxruntime/capi/onnxruntime_inference_collection.py", line 312, in _create_inference_session
    sess.initialize_session(providers, provider_options)
onnxruntime.capi.onnxruntime_pybind11_state.Fail: [ONNXRuntimeError] : 1 : FAIL : The node is not placed on any Execution
Provider. Identity(14) (node Identity_19).
```
I'll try to debug this further.

**Comment by garymm** [comment]
Failures were due to using onnxruntime 1.7 instead of 1.8. Tests pass with 1.8.
LGTM!

**Pull request merged**